stdlib/LinearAlgebra/src/adjtrans.jl

```diff
@@ -126,6 +128,35 @@ axes(v::AdjOrTransAbsVec) = (Base.OneTo(1), axes(v.parent)...)
 axes(A::AdjOrTransAbsMat) = reverse(axes(A.parent))
 IndexStyle(::Type{<:AdjOrTransAbsVec}) = IndexLinear()
 IndexStyle(::Type{<:AdjOrTransAbsMat}) = IndexCartesian()
+
+
+# MemoryLayout of transposed and adjoint matrices
+struct ConjLayout{ML<:MemoryLayout} <: MemoryLayout
+    layout::ML
+end
+conjlayout(::Type{<:Complex}, M::ConjLayout) = M.layout
+conjlayout(::Type{<:Complex}, M::MemoryLayout) = ConjLayout(M)
```

Comment:
These need to support `Any[]` arrays.  They should assume that the elements of an `Any[]` array are going to be conjugated.  Perhaps this should just specialize `::Type{<:Real}` and assume everything else is a `ConjLayout`.

Reply:
Is there a reason that the conjugate of an `Any` array can't just return `UnknownLayout()`?